fix(#533) Replace Park and Ride icon in itinerary

jpidelatorre opened the pull request:
- Use the same logic for the park and ride composite icon as the one used for bike and vehicle icons.
- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

jpidelatorre committed:
fix(#533) Replace Park and Ride icon in itinerary

- Use the same logic for the park and ride composite icon as the one used for bike and vehicle icons.
- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

jpidelatorre committed:
fix(#533) Replace Park and Ride icon in itinerary

- Use the same logic for the park and ride composite
icon as the one used for bike and vehicle icons.

- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

jpidelatorre committed:
fix(#533) Replace Park and Ride icon in itinerary

- Use the same logic for the park and ride composite icon as the one used for bike and vehicle icons.
- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

jpidelatorre committed:
fix(#533) Replace Park and Ride icon in itinerary

- Use the same logic for the park and ride composite icon as the one used for bike and vehicle icons.
- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

jpidelatorre commented:
## Result

![image](https://user-images.githubusercontent.com/104569/121052278-38f51e80-c788-11eb-8e34-e9a745a9f05d.png)

![image](https://user-images.githubusercontent.com/104569/121052346-4ad6c180-c788-11eb-8324-2a1685b5b45b.png)

leonardehrenfried commented:
In your example the blue upper icon should be the bus.

leonardehrenfried commented:
Compare with bike and ride:
![image](https://user-images.githubusercontent.com/151346/121053180-5f807d00-c7bb-11eb-846d-b85b0864107b.png)

leonardehrenfried commented:
In the summary it doesn't need to show a the car twice, just like in the bike view.

![Screenshot from 2021-06-07 18-09-29](https://user-images.githubusercontent.com/151346/121053341-876fe080-c7bb-11eb-8ed8-f80836a3ecd1.png)

jpidelatorre commented:
I was not filtering the steps correctly. It's done now.

![image](https://user-images.githubusercontent.com/104569/121061947-81193e80-c792-11eb-8eb7-cbf8c68aa48e.png)

jpidelatorre commented:
> In the summary it doesn't need to show a the car twice, just like in the bike view.
> 
> ![Screenshot from 2021-06-07 18-09-29](https://user-images.githubusercontent.com/151346/121053341-876fe080-c7bb-11eb-8ed8-f80836a3ecd1.png)

Something like this?
![image](https://user-images.githubusercontent.com/104569/121062134-ba51ae80-c792-11eb-85c3-1eb79330d8f3.png)

leonardehrenfried commented:
what happens when it's just driving? https://dev.stadtnavi.eu/reiseplan/Nufringen%3A%3A48.6224388%2C8.8881021/Herrenberg%3A%3A48.595557%2C8.8674715/parkAndRide?time=1623092264

jpidelatorre commented:
> what happens when it's just driving? https://dev.stadtnavi.eu/reiseplan/Nufringen%3A%3A48.6224388%2C8.8881021/Herrenberg%3A%3A48.595557%2C8.8674715/parkAndRide?time=1623092264

It looks like this:

![image](https://user-images.githubusercontent.com/104569/121074173-79ad6180-c7a1-11eb-84c3-9c05d385d034.png)

leonardehrenfried commented:
And the icon on top?

jpidelatorre commented:
It looks like this:

![image](https://user-images.githubusercontent.com/104569/121098648-521bc080-c7c4-11eb-9c91-ebd4ec739877.png)

leonardehrenfried commented:
Then we have no way to tell the difference between drive and park + walk. What does it look like if you put the walk symbol in the background?

leonardehrenfried commented:
Can you please put the walking icon behind the car when it's just car + walk?

jpidelatorre commented:
I could (same with bike), but there's always a walk before a train or bus. That would mean the other icons would never appear.

leonardehrenfried commented:
Well, can't you modify the logic to check if it's _only_ drive and walk?

jpidelatorre commented:
All the way up to the destination? Sure.

leonardehrenfried commented:
Yes, you can get all the modes from the itinerary and it it's just WALK and CAR then display the walking person icon. Maybe it looks ugly though so fake it before writing the logic.

jpidelatorre commented:
It looks like this:

![image](https://user-images.githubusercontent.com/104569/121315008-f04b7b80-c8d5-11eb-8446-bc4af60aa240.png)

leonardehrenfried commented:
It would probably look nicer with a different coloured background like the rest but we can ask the designers for that. Let's do it.

leonardehrenfried commented:
One question: this is only for the Park and Ride button, right?

jpidelatorre commented:
Yes, only for park and ride.

jpidelatorre committed:
fix(#533) Change car park icon in itinerary

Changed the car park icon and the color behind it.

jpidelatorre committed:
fix(#533) Change car park icon in itinerary

Changed the car park icon and the color behind it.

jpidelatorre commented:
As per @hbruch request, here's how the new icon looks like:

![image](https://user-images.githubusercontent.com/104569/121331430-dbc2af80-c8e4-11eb-91ef-64b72dbf287f.png)

leonardehrenfried commented on app/component/SummaryRow.js:
```diff
@@ -72,6 +72,7 @@ export const RouteLeg = ({
   isTransitLeg,
   fitRouteNumber,
   withBicycle,
+  subIcon,
```
Is the subicon required in this component?

leonardehrenfried commented on app/component/RouteNumber.js:
```diff
@@ -76,6 +79,9 @@ function RouteNumber(props, context) {
             className="itinerary-icon_with-bicycle"
           />
         )}
+        {subIcon && (
```
what does this do?

leonardehrenfried commented:
The colours of train trips are also messed up:
### Before
![Screenshot from 2021-06-09 13-23-02](https://user-images.githubusercontent.com/151346/121346031-dc2f6a80-c925-11eb-8fcd-e60becabb3df.png)
### After
![Screenshot from 2021-06-09 13-23-08](https://user-images.githubusercontent.com/151346/121346027-db96d400-c925-11eb-8148-bee734bcbcb8.png)

leonardehrenfried commented:
![Screenshot from 2021-06-09 13-25-53](https://user-images.githubusercontent.com/151346/121346295-3d573e00-c926-11eb-8b14-e542ddc9af85.png)

This display is also incorrect. Please calculate the distance for the driving part.

jpidelatorre commented:
@leonardehrenfried  I pushed some corrections. I think every point is solved with that.

**This is what I see in the icons. Is this correct?**

![image](https://user-images.githubusercontent.com/104569/121471828-4e3b9a00-c98e-11eb-8b82-b63c855073e9.png)

---

**Is this the correct calculation?**

![image](https://user-images.githubusercontent.com/104569/121472108-ba1e0280-c98e-11eb-9559-0de9d23ca456.png)

leonardehrenfried commented:
![Screenshot from 2021-06-10 11-11-44](https://user-images.githubusercontent.com/151346/121498400-ab5e3c80-c9dc-11eb-8afc-c940c73d415e.png)

Car is missing from P+R.

jpidelatorre committed:
Rectified icon color

Color was not being correctly set. Additionally, moved the width of the icon to the stylesheet.

jpidelatorre commented:
![image](https://user-images.githubusercontent.com/104569/121502471-33791d80-c9ae-11eb-9578-03d9311ec276.png)

Fixed. I had removed the color for the icon from the markup without adding it back in the stylesheet. Now it is correct.

leonardehrenfried commented:
Thanks.
